sql/util/requests.sql

```diff
@@ -178,21 +167,34 @@ _cdn_provider: $._cdn_provider,
 _gzip_save: $._gzip_save,
 type: prettyType,
 ext: ext,
-format: getFormat(prettyType, mimeType, ext)
-    };
+format: getFormat(prettyType, mimeType, ext,),
+protocol: $._protocol,
+pushed: $._was_pushed,
+tls_version: $._tls_version,
+tls_cipher_suite: $._tls_cipher_suite,
+cert_issuer: securityDetails.issuer,
+cert_keyexchange: securityDetails.keyExchange,
+cert_cipher: securityDetails.cipher,
+cert_protocol: securityDetails.protocol
 
+    };
     return summary;
   } catch (e) {
     return e
   }
 """;
 
 SELECT
-  CAST('2020-08-01' AS DATE) AS date,
+  CAST('2021-07-01' AS DATE) AS date,
   _TABLE_SUFFIX AS client,
   page,
-  url,
-  getSummary(payload).*, --  noqa: PRS, L013
+  rank,
+  req.url as url,
+  getSummary(payload).*,
+  JSON_EXTRACT(payload, "$.request.headers") AS request_headers,
+  JSON_EXTRACT(payload, "$.response.headers") AS response_headers,
   payload
 FROM
-  `httparchive.requests.2020_08_01_*`
+  `httparchive.requests.2021_07_01_*` req
+LEFT JOIN (SELECT DISTINCT url, rank FROM `httparchive.summary_pages.2021_07_01_*`) rank_data
+ON req.page = rank_data.url
```

Comment:
I think we can omit `DISTINCT` since the summary_pages dataset only contains unique pages. Also reformatting and aliasing `url` to avoid table aliases.

```suggestion
  `httparchive.requests.2021_07_01_*
LEFT JOIN (
  SELECT
    url AS page,
    rank
  FROM
    `httparchive.summary_pages.2021_07_01_*`)
USING (page)
```

Reply:
That query is collecting the distinct url, rank combination from both desktop and mobile tables.    elimating this would result in many sites having two entries in that joined table (which would artifically increase the size of the results).

Reply:
Oh, shouldn't we be adding `client` to this subquery and using the rank for that `client`? Are the ranks always the same for `desktop` and `mobile`?

Reply:
Sorry @paulcalvano is right that my suggestion doesn't work.

Reply:
Looks like it is the same rank as this returns zero rows:

```sql
SELECT
    d.url,
    d.rank as desktop_rank,
    m.rank AS mobile_rank
FROM
    `httparchive.summary_pages.2021_07_01_desktop` d,
    `httparchive.summary_pages.2021_07_01_mobile` m
WHERE 
    d.url = m.url AND 
    d.rank != m.rank
```

Not sure we should assume that though, in case it ever changes.

Any harm to adding the client to this?
And then also don't need the `DISTINCT`?

Reply:
I added the client in (and also to the JOIN) just in case, and removed the DISTINCT.